src/dpl/src/Opendp.cpp

```diff
@@ -456,19 +447,37 @@ int Opendp::padRight(dbInst* inst) const
   return 0;
 }
 
+int Opendp::paddedWidth(const Cell* cell, int site_width) const
+{
+  return cell->width_ + (padLeft(cell) + padRight(cell)) * site_width;
+}
+
 int Opendp::paddedWidth(const Cell* cell) const
 {
-  return cell->width_ + (padLeft(cell) + padRight(cell)) * site_width_;
+  int site_width = getSiteWidth(cell);
+  return cell->width_ + (padLeft(cell) + padRight(cell)) * site_width;
+}
+
+int Opendp::gridPaddedWidth(const Cell* cell, int site_width) const
+{
+  return divCeil(paddedWidth(cell), site_width);
 }
```

Comment:
I see your point, but when we support multi-site widths, we might potentially do a logarithmic look-up on the grid_layers_ or other similar data structures. 

The lookup logic inside getSiteWidth is currently an O(1) since we don't support multi-site-widths yet, so your comment makes perfect sense, but once we replace it, doing the lookup once beforehand and storing the site-width for all subsequent calls that require the value might be a good optimisation IMO.